gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py

```diff
@@ -0,0 +1,115 @@
+    ]:
+        vep_ht = vep_ht.annotate(
+            vep=vep_ht.vep.annotate(
+                **{
+                    consequence: vep_ht.vep[consequence].map(
+                        lambda x: x.drop("minimised")
+                    )
+                }
+            )
+        )
+    return vep_ht
+
+
+def replace_oth_with_remaining(ht: hl.Table) -> hl.Table:
+    """
+    Replace 'oth' with 'remaining' in Global fields of a Table.
+
+    .. note::
+        This function is used to rename ancestry group fields to match v4 exomes.
+        Which means that all "oth" will be changed to "remaining".
+        2 Global fields are renamed: freq_meta (in the value of the key "pop") and freq_index_dict (in the keys).
+
+    :param ht: release sites Table to be modified
+    :return: release sites Table with 'oth' replaced with 'remaining'
+    """
+    # in freq_meta, type: array<dict<str, str>>, the value of key "pop" is
+    # changed from 'oth' to 'remaining'.
+    ht = ht.annotate_globals(
+        freq_meta=ht.freq_meta.map(
+            lambda d: d.map_values(lambda x: x.replace("oth", "remaining"))
+        )
+    )
+
+    # in freq_index_dict, type: dict<str, int>, the keys containing 'oth'
+    # inside are changed to 'remaining'.
+    oldkeys = hl.eval(ht.freq_index_dict.keys())
+    newkeys = [s.replace("oth", "remaining") for s in oldkeys]
+    vals = hl.eval(ht.freq_index_dict.values())
+    freq_index_dict_new = {k: v for k, v in zip(newkeys, vals)}
+    ht = ht.annotate_globals(freq_index_dict=freq_index_dict_new)
+    return ht
+
+
+# TODO: drop old_locus, old_alleles, seems to be already dropped in v3.1.4 release table
+# TODO: rename ancestry group fields to match v4 exomes
+# TODO: drop missing fields from VEP 105 annotations
+# TODO: merge vep_ht with v3.1.4 release table
+# TODO: rerun inbreeding coefficient with callstats instead of GT calls
+# TODO: annotate with newest dbsnp release
+# TODO: add Pangolin prediction
+# TODO: add zoonomia constraint scores
+
+
+def main(args):
+    """Script to generate release sites ht for v4 genomes."""
+    logger.info("Loading release table of v3.1.4...")
+    # ht = release_sites(public=True).versions["3.1.4"].ht()
+    ht = hl.read_table(
+        "gs://gnomad/release/3.1.4/ht/genomes/gnomad.genomes.v3.1.4.sites.ht"
+    )
+
+    logger.info("Replacing ancestry group 'oth' with 'remaining'...")
+    ht = replace_oth_with_remaining(ht)
+
+    logger.info("Loading VEP-annotated table...")
+    vep_ht = get_vep(version=CURRENT_VERSION, data_type="genomes").ht()
+
+    logger.info("Removing missing VEP fields...")
+    vep_ht = remove_missing_vep_fields(vep_ht)
+
+    logger.info("Loading dbsnp table...")
+    dbsnp_ht = dbsnp.ht().select("rsid")
+
+    logger.info("Loading pangolin table...")
+    pangolin_ht = hl.read_table(
+        "gs://gnomad/v4.0/annotations/genomes/gnomad.genomes.v4.0.pangolin.ht"
+    )  # TODO: make Pangolin a versioned resource
```

Comment:
@KoalaQin -- this looks incomplete -- perhaps missing a commit? Nothing happens with the dbsnp or pangolin tables, looks like they are just read in. Would you mind adding it before I do a full review?

Reply:
No, these are for other steps, because I will put them in one file. Could you finish the pending two PRs that are independent functions？

Reply:
In general, its best to have commits that are going into main be complete functions. These lines should be dropped from this module since it adds confusion. What is the second PR that is independent? I only see #359 assigned to me

Reply:
Yes, it's #359. Could you start that one first? I can delete the "main" part from #363. 
#546 in gnomad_methods is for VRS fields.

Reply:
Yup! Sorry -- just to be clear https://github.com/broadinstitute/gnomad_methods/pull/546  shows VEP consequence and header updates. Is there another I should be looking at for VRS?

Reply:
Nope, I only got these 3 PRs now and it's all on you. Thanks a lot!